taskrc/ast/taskrc.go

```diff
@@ -43,6 +44,31 @@ func (t *TaskRC) Merge(other *TaskRC) {
 	t.Remote.Timeout = cmp.Or(other.Remote.Timeout, t.Remote.Timeout)
 	t.Remote.CacheExpiry = cmp.Or(other.Remote.CacheExpiry, t.Remote.CacheExpiry)
 
+	// Merge Trust lists - combine both lists with other's entries taking precedence
+	// Remove duplicates by using a map
+	if len(other.Remote.Trust) > 0 {
+		seen := make(map[string]bool)
+		merged := []string{}
+
+		// Add other's hosts first
+		for _, host := range other.Remote.Trust {
+			if !seen[host] {
+				seen[host] = true
+				merged = append(merged, host)
+			}
+		}
+
+		// Then add base's hosts that aren't duplicates
+		for _, host := range t.Remote.Trust {
+			if !seen[host] {
+				seen[host] = true
+				merged = append(merged, host)
+			}
+		}
+
+		t.Remote.Trust = merged
+	}
+
```

Comment:
Since the order doesn’t matter (a host is trusted regardless of its position in the slice), we can simplify it to:


```suggestion
	if len(other.Remote.Trust) > 0 {
		merged := slices.Concat(other.Remote.Trust, t.Remote.Trust)
		slices.Sort(merged)
		t.Remote.Trust = slices.Compact(merged)
	}

```